packages/actor-rdf-parse-html-script/lib/ActorRdfParseHtmlScript.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  ActorRdfParseFixedMediaTypes,
+  IActionRdfParse,
+  IActionRootRdfParse,
+  IActorOutputRootRdfParse,
+  IActorRdfParseFixedMediaTypesArgs,
+  IActorRdfParseOutput,
+  IActorTestRootRdfParse,
+} from "@comunica/bus-rdf-parse";
+import {ActionContext, Actor, Mediator} from "@comunica/core";
+import {Readable} from "stream";
+
+/**
+ * A HTML script RDF Parse actor that listens on the 'rdf-parse' bus.
+ *
+ * It is able to extract and parse any RDF serialization from HTML files
+ * and announce the presence of them by media type.
+ */
+export class ActorRdfParseHtmlScript extends ActorRdfParseFixedMediaTypes {
+
+  public mediatorRdfParse: Mediator<Actor<IActionRootRdfParse, IActorTestRootRdfParse,
+    IActorOutputRootRdfParse>, IActionRootRdfParse, IActorTestRootRdfParse, IActorOutputRootRdfParse>;
+
+  constructor(args: IActorRdfParseFixedMediaTypesArgs) {
+    super(args);
+  }
+
+  public async runHandle(action: IActionRdfParse, mediaType: string, context: ActionContext):
+    Promise<IActorRdfParseOutput> {
+
+    const quads = new Readable({objectMode: true});
+
+    quads._read = async () => {
+
+      // Stringify HTML input
+      const htmlString: string = await require('stream-to-string')(action.input);
```

Comment:
This should not be needed.
`htmlparser2` accepts streams AFAIK, so you should be able to pipe `action.input` into the parser directly. This will lead to much better performance, as stuff can happen asynchronously.